polkadot/node/core/pvf/src/host.rs

```diff
@@ -203,11 +204,14 @@ impl Config {
 /// The future should not return normally but if it does then that indicates an unrecoverable error.
 /// In that case all pending requests will be canceled, dropping the result senders and new ones
 /// will be rejected.
-pub async fn start(config: Config, metrics: Metrics) -> (ValidationHost, impl Future<Output = ()>) {
+pub async fn start(
+	config: Config,
+	metrics: Metrics,
+) -> SubsystemResult<(ValidationHost, impl Future<Output = ()>)> {
 	gum::debug!(target: LOG_TARGET, ?config, "starting PVF validation host");
 
 	// Run checks for supported security features once per host startup. Warn here if not enabled.
-	let security_status = security::check_security_status(&config).await;
+	let security_status = security::check_security_status(&config).await?;
```

Comment:
Wait a second. This looks worse, with regards to the issue. If we can't check, we can now no longer do any validation, despite being not yet strict in enforcing security? Am I missing something?

Reply:
Right now the only possible error should never happen (https://github.com/paritytech/polkadot-sdk/blob/2e2a75ff81/polkadot/node/core/pvf/src/worker_intf.rs#L189-L196). But I could have done this better. Thanks!